packages/kbn-journeys/services/page/kibana_page.ts

```diff
@@ -83,6 +83,18 @@ export class KibanaPage {
     });
   }
 
+  async waitForChartsSuggestions(count: number) {
+    await this.page.waitForFunction(
+      function renderCompleted(args: { count: number }) {
+        const renderingItems = Array.from(
+          document.querySelectorAll('button[data-test-subj="lnsSuggestion"]')
+        );
+        return renderingItems.length === args.count;
+      },
+      { count }
+    );
```

Comment:
I recently discovered that because of CSP enabled, `page.waitForFunction` will be detected as `unsafe-eval` and basically won't work on Cloud . 

To make it work in both local (CSP is off) and Cloud deployments/projects we have to use playwright API + retry service:

smth like
```
await this.retry.waitFor(
  `rendering of ${count} suggestions is completed`,
  async () => {
      const renderingItems = await this.page.$$(button[data-test-subj="lnsSuggestion"]');
      return renderingItems.length === args.count;
  }

)
```

Reply:
@dmlemeshko thanx for letting me know! Here it is https://github.com/elastic/kibana/pull/166808/commits/983c25859eb2b6848e8f6f1c41a7c4e0f21db732